### PR TITLE
Support zipped index folder (stdlib)

### LINKED ISSUE #8: Utilize Zip format for index folder
The index is a directory with 250,000+ files, several GB in size. However, it compresses well, therefore consider a single Zip archive or annual/monthly Zip archives. Note, a monthly Zip archive can be updated more efficiently.

### PR DESCRIPTION
**Pull request by forman**
The NetCDF Kerchunk index can now also be Zip archive. To use a Zip archive, pass an index path to the `nckcidx` CLI or to the `NcKcIndex`class whose filename has a `.zip` extension.

Closes #8

### CODE REVIEW

File: xcube_smos/nckcindex/nckcindex.py [hunk 1]
```diff
@@ -134,44 +135,36 @@ def create(
...
         """Open the given index at *index_path*.
 
         :param index_path: The index path or URL.
-        :param index_protocol: The index protocol.
-            If not provided, it will be derived from *index_path*.
-        :param index_storage_options: Optional storage options for accessing
-            the filesystem of *index_path*.
-            See Python fsspec package spec for the used index protocol.
+        :param mode: Open mode, must be either "w" or "r".
+            Defaults to "r".
```
Comment (pont-us) [inline xcube_smos/nckcindex/nckcindex.py:152]: Contradicts the implementation below, which allows "a" in addition to "w" and "r".

**Review by pont-us — APPROVED** [review]
I like it, very clear implementation. A few minor requests / comments, but approving since I don't need to review again.

File: README.md [hunk 2]
```diff
@@ -9,43 +9,41 @@ Level-2 multi-level datasets.
 ## Index Format
 
 Consider some `<source-path>` that points into a NetCDF file archive.
```
[the remark this answers is not in the document]
Comment (forman) [reply README.md:11]: `<source-prefix>` is wrong. I'll fix it.

[the remark this answers is not in the document]
Comment (forman) [reply README.md:11]: Fixed.

**Pull request merged**